Source/RxCocoa/UIScrollView+reachedBottom.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  UIScrollView+reachedBottom.swift
+//  RxSwiftExt
+//
+//  Created by Anton Nazarov on 09/05/2019.
+//  Copyright © 2019 RxSwift Community. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+
+public extension Reactive where Base: UIScrollView {
+    /**
+     Shows if the bottom of the UIScrollView is reached.
+     - parameter offset: A threshhold indicating the bottom of the UIScrollView.
+     - returns: ControlEvent that emits when the bottom of the base UIScrollView is reached.
+     */
+    func reachedBottom(offset: CGFloat = 0.0) -> ControlEvent<Void> {
+        let source = contentOffset.map { contentOffset in
+            let visibleHeight = self.base.frame.height - self.base.contentInset.top - self.base.contentInset.bottom
+            let y = contentOffset.y + self.base.contentInset.top
+            let threshold = max(offset, self.base.contentSize.height - visibleHeight)
+            return y >= threshold
+        }
```

Comment:
You would need `distinctUntilChanged` here to remove duplicate states.

Reply:
I don't want to remove dublicate states, because scrollView can reach bottom more than once

Reply:
Don't think so - you want to know when it arrives, not the 50 times you'd cross the boundary, right?

Once you reach the bottom you can't "reach the bottom again" until you go up. Did you try this when you say it causes strange behavior ?

Reply:
Yes i can. I show 20 cells. reach bottom, load another 20 cells from backend and than reach bottom again

Reply:
But then the offset would change since you added more cells, so it would be like "scrolling" and would change back to false. At least that's what it seems to me, I didn't try it.

Reply:
I thought the offset would change but it actually wouldn't change by just adding cells, I imagine. I don't have the time to try this - @jdisho did you test this by any chance?

Reply:
Sure I did @freak4pc and works as expected.

Reply:
Ok, @MortyMerr - did you actually test your hypothesis? I wrote something like this as well and remember adding `distinctUntilChanged` or it was too sporadic.

Reply:
@freak4pc I just didn't get your point from the first time. Yes, it will work as you described. I added `disntinctUntilChanged`